### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by crandmck**
Per https://github.com/strongloop/loopback.io/issues/160

@Amir-61

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,11 +1,414 @@
...
+      "geocode": ["street", "city", "zipcode"]
+    }
+  }]
+}
+...
+```
+
+For a REST data source, you can define an array of  _operation_ objects to specify the REST API mapping. Each operation object can have the following two properties:
+
+* template: See how to define a custom method template [below](#defining-a-custom-method-using-a-template).
```
Comment (Amir-61) [inline README.md]: Nitpick: Shouldn't it start with capital letter `T`: `Template:...`?

File: README.md [hunk 2]
```diff
@@ -1,11 +1,414 @@
...
+* functions: An object that maps a JavaScript function to a list of parameter names.
```
Comment (Amir-61) [inline README.md]: Nitpick: Shouldn't it start with capital letter `F`: `Functions:...`?

**Comment by Amir-61** [comment]
Also please squash your commits into a single commit. Thanks!

**Comment by Amir-61** [comment]
@crandmck Thanks for updating documentation 👍 
I just wanted to bring up that we always squash our commits into one single commit if the commits are related and in the same category of changes... For instance now I see you have pushed 6 commits with commits message of `Update README.md`; next time could you please squash them as one single commit? You can do that easily by using the following command line if you are pushing against `master` branch:
```
git rebase -i origin/master
```
And then you can interactively squash the commits, reword them,...

Last but not least after landing PRs we delete our feature branch as well.

Thanks for documentation 🙇

**Comment by Amir-61** [comment]
ummm I guess you used `squash and merge` option in GITHUB UI and the actual commit (https://github.com/strongloop/loopback-connector-rest/commit/dd259a2929f5007a7df86f6c6a3c9c9dd324cdd5) on `master` is already a squashed of all your commits,... So you may ignore my previous comment :-)

**Comment by crandmck** [comment]
Yep, it's much easier to squash and merge with the button.  Sorry I forgot to delete the branch.  I usually do, but it was an oversight.

**Pull request merged**